File: 1-helloworld/helloworld-mpi.c
#include <stdio.h>

#include <mpi.h>

void main()
{
    int rank, process;

    MPI_Init(NULL, NULL);
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    MPI_Comm_size(MPI_COMM_WORLD, &process);

    printf("Hello world! from process %d\n", rank);

    MPI_Finalize();
}

File: 1-helloworld/helloworld-mpi.f90
program helloworld
    implicit none
    include 'mpif.h'

    INTEGER :: rank
    INTEGER :: size
    INTEGER :: ierror
    INTEGER :: tag
    INTEGER :: status(MPI_STATUS_SIZE)

    call MPI_INIT(ierror)
    call MPI_COMM_SIZE(MPI_COMM_WORLD, size, ierror)
    call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierror)

    print *, 'Hello world! from process ', rank

    call MPI_FINALIZE(ierror)
    
end program helloworld